internal/logs/workload.go

```diff
@@ -0,0 +1,199 @@
+package logs
+
+import (
+	"context"
+	"time"
+
+	"git.sr.ht/~spc/go-log"
+	"github.com/project-flotta/flotta-device-worker/internal/workload"
+	"github.com/project-flotta/flotta-device-worker/internal/workload/podman"
+	"github.com/project-flotta/flotta-operator/models"
+)
+
+type transportLog struct {
+	transport transport
+	log       *FifoLog
+	cancel    context.CancelFunc
+}
+
+func (t *transportLog) Start() {
+	ctx, cancel := context.WithCancel(context.Background())
+	t.cancel = cancel
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			default:
+				entry, err := t.log.ReadLine()
+				if err != nil {
+					log.Error("Error on read log info", err)
+				}
+				if entry == nil {
+					time.Sleep(elapsedTime)
+				} else {
+					err = t.transport.WriTeLog(entry)
+					if err != nil {
+						// cannot send, pushed again to the queue.
+						err := t.log.Write(entry)
+						if err != nil {
+							log.Error("cannot return log entry to the buffer log after transport failed", err)
+						}
+					}
+				}
+			}
+		}
+	}()
+}
+
+func (t *transportLog) Stop() {
+	if t.cancel != nil {
+		t.cancel()
+	}
+}
+
+type WorkloadsLogsTarget struct {
+	transports      map[string]transportLog
+	workloads       []*WorkloadWriter
```

Comment:
There is a lot of map-like operations against that slice (getting by name, removing by name) - can it be a map and have O(1) instead of O(n) for these operations?

Reply:
I did as O(1) but I need to make complex structs, at the end, the number of workloads will be small(<10) so I think that it's much better dealing in this way than in the other way. 

This is my personal opinion, and by the kind of devices that we're supposed to manage

Reply:
IMO [these](https://github.com/project-flotta/flotta-device-worker/pull/101/files#diff-b5eb0d3de573bbf6fc6b46b3b42057737eb1b7c236b49b381231b7198e6aac9bR85-R94) 9 lines can be replaced with 3
and actually, why doesn't it use https://github.com/project-flotta/flotta-device-worker/pull/101/files#diff-b5eb0d3de573bbf6fc6b46b3b42057737eb1b7c236b49b381231b7198e6aac9bR97 ?

Reply:
At the end, you need to loop over the deletetarget, but I'll change.